internal/model/model.go

```diff
@@ -1388,6 +1388,83 @@ func (m *Model) RemoteLocalVersion(folder string) int64 {
 	return ver
 }
 
+func (m *Model) GetDirectoryTree(folder, prefix string, levels int, dirsonly bool) map[string]interface{} {
```

Comment:
Looks sane enough, but could we have a test just for the sake of it? I isn't completely obvious to me that the stack popping mechanism etc works in all cases like `foo/bar/baz` followed directly by `something/else/entirely` and so on...